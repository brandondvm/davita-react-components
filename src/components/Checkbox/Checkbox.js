import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Input from '../Input/Input';

const Checkbox = ({
  id,
  name,
  label,
  checked,
  textInput,
  register,
  watchFields,
}) => {
  const [isChecked, setIsChecked] = useState(checked);
  const toggleCheckboxState = () => setIsChecked(!isChecked);

  return (
    <div className="checkbox-component">
      <input
        id={id}
        aria-checked={isChecked}
        defaultChecked={isChecked}
        ref={register}
        name={name}
        aria-label={label}
        className="visually-hidden"
        onChange={toggleCheckboxState}
        type="checkbox"
      />
      <label
        htmlFor={id}
        className={`checkbox-component_label${isChecked ? ' checked' : ''}`}
      >
        {label}
      </label>
      {watchFields && watchFields[textInput?.watching] && (
        <Input
          id={textInput.id}
          placeholder={textInput.placeholder}
          name={textInput.name}
          type={textInput.type}
          register={register(textInput.validation)}
        />
      )}
    </div>
  );
};

Checkbox.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  label: PropTypes.string,
  checked: PropTypes.bool,
  textInput: PropTypes.object,
  register: PropTypes.func,
};

export default Checkbox;
