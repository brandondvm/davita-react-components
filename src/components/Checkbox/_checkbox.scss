.checkbox-component {
  margin-bottom: 15px;

  &_label {
    cursor: pointer;
    display: flex;
    align-items: center;
    &::before {
      content: "";
      border-radius: 4px;
      border: 1px solid $mediumgray;
      box-sizing: border-box;
      display: block;
      font: normal normal normal #{14px}/#{1px} $icon-font;
		  height: 20px;
      width: 20px;
      margin-right: 10px;
    }

    &.checked {
      &::before {
        content: "\2713";
        background-color: $lightgreen;
        border-radius: 4px;
        border: 1px solid $lightgreen;
        color: $white;
        font-size: 10px;
        font-weight: bold;
        height: 20px;
        line-height: 15px;
        width: 20px;
        display: flex;
        align-items: center;
        justify-content: center;
      }
    }
  }

  label {
    color: $onyx;
    font-size: 14px;
    margin: 0;
  }
}
