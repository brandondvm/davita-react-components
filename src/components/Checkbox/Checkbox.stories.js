import React from 'react';

import Checkbox from './Checkbox';

export default {
  title: 'Checkbox',
  component: Checkbox,
  argTypes: {
    label: 'string'
  },
};

const Template = (args) => <div className="storybook-wrapper"><Checkbox {...args} /></div>;

export const Default = Template.bind({});
Default.args = {
  id: 'checkbox1',
  name: 'checkbox1',
  label: "Option 1"   
};

export const Active = Template.bind({});
Active.args = {
  id: 'checkbox1',
  name: 'checkbox1',
  label: "Option 1",
  checked: true
};
