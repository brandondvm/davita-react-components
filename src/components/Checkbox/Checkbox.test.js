import enzyme from 'enzyme';
import React from 'react';

import Checkbox from './Checkbox';

const mockProps = {
  id: "option1",
  label: "option1",
  name: "option1",
  value: "option1",
  checked: false,
};

function setup(passedProps = {}) {
  const updatedProps = {
    ...mockProps,
    ...passedProps,
  };

  return enzyme.shallow(<Checkbox { ...updatedProps } />);
}

describe('Checkbox Component', () => {
  test('Component renders correctly', () => {
    const wrapper = setup();
    expect(wrapper.get(0)).toMatchSnapshot();
  });

  test('Component should render props correctly on DOM', () => {
    const wrapper = setup();
    const checkboxLabel = wrapper.find('.checkbox-component_label');
    expect(checkboxLabel.text()).toEqual(mockProps.label);
  });

  test('Component should render correctly checkbox checked', () => {
    const wrapper = setup({ checked: true });
    const inputLabelEl = wrapper.find('.checkbox-component_label');
    expect(inputLabelEl.hasClass('checked')).toBeTruthy();
  });

  test('Component should render correctly checkbox unchecked', () => {
    const wrapper = setup();
    const inputLabelEl = wrapper.find('.checkbox-component_label');
    expect(inputLabelEl.hasClass('checked')).toBeFalsy();
  });

  test('Component should toggle checkbox status', () => {
    const wrapper = setup();
    let inputLabelEl = wrapper.find('.checkbox-component_label');
    expect(inputLabelEl.hasClass('checked')).toBeFalsy();

    wrapper.find('.visually-hidden').simulate('change');

    inputLabelEl = wrapper.find('.checkbox-component_label');
    expect(inputLabelEl.hasClass('checked')).toBeTruthy();
  });
});
