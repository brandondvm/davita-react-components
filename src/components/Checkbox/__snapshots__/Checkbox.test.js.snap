// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component Component renders correctly 1`] = `
<div
  className="checkbox-component"
>
  <input
    aria-checked={false}
    aria-label="option1"
    className="visually-hidden"
    defaultChecked={false}
    id="option1"
    name="option1"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="checkbox-component_label"
    htmlFor="option1"
  >
    option1
  </label>
</div>
`;
